Mathlib/Order/KrullDimension.lean

```diff
@@ -106,15 +179,32 @@ lemma krullDim_eq_of_orderIso (f : α ≃o β) : krullDim α = krullDim β :=
   le_antisymm (krullDim_le_of_strictMono _ f.strictMono) <|
     krullDim_le_of_strictMono _ f.symm.strictMono
 
-lemma krullDim_eq_iSup_height : krullDim α = ⨆ (a : α), height α a :=
-  le_antisymm
-    (iSup_le fun i ↦ le_iSup_of_le (i ⟨i.length, lt_add_one _⟩) <|
-    le_sSup ⟨⟨_, fun m ↦ ⟨i m, i.strictMono.monotone <| show m.1 ≤ i.length by omega⟩,
-      i.step⟩, rfl⟩) <|
-    iSup_le fun a ↦ krullDim_le_of_strictMono Subtype.val fun _ _ h ↦ h
-
 @[simp] lemma krullDim_orderDual : krullDim αᵒᵈ = krullDim α :=
   le_antisymm (iSup_le fun i ↦ le_sSup ⟨i.reverse, rfl⟩) <|
     iSup_le fun i ↦ le_sSup ⟨i.reverse, rfl⟩
 
+/--
+The Krull dimension is the supremum of the element's heights.
+
+If `α` is `Nonempty`, then `krullDim_eq_iSup_height_of_nonempty`, with the coercion from
+`ℕ∞` to `WithBot ℕ∞` outside the supremum, can be more convenient.
+-/
+lemma krullDim_eq_iSup_height : krullDim α = ⨆ (a : α), (height a : WithBot ℕ∞) := by
```

Comment:
Can we also have (not necessarily in this PR)
* `krullDim α = ⨆ a : α, coheight a`
* `krullDim α = ⨆ a : α, height a + coheight a`

Reply:
The first should be in the full PR (but I'll check).

The second is nice, will add there

Reply:
Yes, the first one already is in #15524 (all lemmas have their coheight dual), and I added the second one.